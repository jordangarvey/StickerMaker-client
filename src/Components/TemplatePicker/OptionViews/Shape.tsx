import React from "react";
import { StyleSheet, TouchableOpacity } from "react-native";

import colours from "../../../Data/colours";
import shapes from "../../../Data/shapes";
import { useAppContext } from "../../../State/AppContext";


import * as C from "../../../Global/Colours";


/**
 * Interface for the Shape component
 */
interface IShapeProps {
	/** The shape this component should display as */
	shape: string;
}

/**
 * Component to render a shape button
 */
function Shape(props: IShapeProps) {
	const [{ values }, dispatch] = useAppContext();

	function onChange() {
		const newValues = values;
		newValues.shape = props.shape;

		dispatch({ payload: newValues, type: "updateValues" });
	}

	let style: object;

	switch(props.shape) {
		case "Capsule":
			style = styles.capsule;
			break;
		case "Rectangle":
			style = styles.rect;
			break;
		case "Square":
			style = styles.square;
			break;
		default:
			throw new Error("Unknown shape");
	}

	return (
		<TouchableOpacity
			onPress={onChange}
			style={
				[
					{ backgroundColor: colours[values.colour] || "#333333" },
					style,
					styles.shape,
					(values.shape === props.shape) && styles.selected
				]
			}
		/>
	);
}

const styles = StyleSheet.create({
	capsule: {
		borderRadius: 80,
		width: 120
	},
	rect: {
		borderRadius: 20,
		width: 120
	},
	selected: {
		borderColor: C.selected,
		borderWidth: 5
	},
	shape: {
		height: 70,
		marginHorizontal: 15,
	},
	square: {
		borderRadius: 20,
		width: 70
	}
});

export default Shape;
