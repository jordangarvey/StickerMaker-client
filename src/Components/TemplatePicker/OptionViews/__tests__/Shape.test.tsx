import React from "react";

import renderer from "react-test-renderer";

import Shape from "../Shape";


describe("The Shape componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		const component = renderer.create(
			<Shape
				shape="rect"
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
