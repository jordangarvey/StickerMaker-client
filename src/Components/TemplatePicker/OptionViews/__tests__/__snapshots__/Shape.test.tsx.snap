// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Shape componentâ€™s snapshot renders as expected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#333333",
      "borderRadius": 20,
      "height": 70,
      "marginHorizontal": 15,
      "opacity": 1,
      "width": 120,
    }
  }
/>
`;
